### PR TITLE
Reconcile router config

### PR DESCRIPTION
**Pull request by lulf**
* Generate router config model identical to what was stored in template
* Add support for (de)serializing router config to JSON
* Remove router config from template
* Add controller for reconciling router config
* Use upstream image from interconnected cloud rather than building our own (removes router module)
* Add better description of address pattern

### CODE REVIEW

File: address-space-controller/src/main/java/io/enmasse/controller/InfraResourceFactory.java [hunk 1]
```diff
@@ -4,10 +4,13 @@
  */
 package io.enmasse.controller;
```
Comment (k-wall) [inline address-space-controller/src/main/java/io/enmasse/controller/InfraResourceFactory.java]: Unnecessary changes?

File: address-space-controller/src/main/java/io/enmasse/controller/RouterConfigController.java [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
+
+        Listener httpsPublic = new Listener();
+        httpsPublic.setHost("0.0.0.0");
+        httpsPublic.setPort(8443);
+        httpsPublic.setSaslPlugin("auth_service");
+        httpsPublic.setSslProfile("ssl_details");
+        httpsPublic.setHttp(true);
+        httpsPublic.setAuthenticatePeer(true);
+        if (routerSpec != null && routerSpec.getPolicy() != null) {
+            httpsPublic.setPolicyVhost("public");
```
Comment (k-wall) [inline address-space-controller/src/main/java/io/enmasse/controller/RouterConfigController.java]: I notice the io.enmasse.controller.common.TemplateParameter now has unused constants.

**Comment by lulf** [comment]
@enmasse-ci run tests

**Comment by lulf** [comment]
@enmasse-ci run tests profile=smoke

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=**.InfraTest

**Comment by lulf** [comment]
@enmasse-ci run tests

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=io.enmasse.systemtest.brokered.web.FirefoxWebConsoleTest,io.enmasse.systemtest.common.monitoring.MonitoringTest

**Comment by lulf** [comment]
@enmasse-ci run tests testcase=io.enmasse.systemtest.common.monitoring.MonitoringTest#testAddressSpaceRules
